components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -186,33 +135,136 @@ private static boolean isHeadRequest(HttpRequest request) {
         return request.method().equals(HEAD);
     }
 
+    private Observable<HttpResponse> sendRequest(HttpRequest request, List<RemoteHost> previousOrigins, int attempt) {
+        if (attempt >= 3) {
```

Comment:
Hi. This is a safety valve to prevent a faulty or otherwise inconsiderate retry policy implementation from indefinitely looping around sending requests.

I agree it is not ideal.

I am welcoming any sensible ideas to address this issue.

Reply:
I think we should at least extract the number as a static final field and add a comment. Maybe add some logging to notify us if the condition occurs?
Maybe make it configurable too, in case someone wants to retry 4 times as well.

Reply:
I have added a static final field for now. Let's make it configurable in an additional increment.